kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/domain/KafkaConsumerGroupMetrics.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.mwam.kafkakewl.metrics.domain
+
+enum ConsumerGroupStatus(val severity: Int):
+  /** Aggregates 2 ConsumerGroupStatuses so that the result is the more severe.
+    */
+  def +(other: ConsumerGroupStatus): ConsumerGroupStatus = if (severity < other.severity) other else this
+
+  case Ok extends ConsumerGroupStatus(1)
+  case Unknown extends ConsumerGroupStatus(2)
+  case Warning extends ConsumerGroupStatus(3)
+  case Error extends ConsumerGroupStatus(4)
+  case MaybeStopped extends ConsumerGroupStatus(5)
+  case Stopped extends ConsumerGroupStatus(6)
+
+end ConsumerGroupStatus
```

Comment:
does json serialization work fine for enums? I never tried.

Reply:
I took a closer look and don't really like it. Implemented my own serialization/deserialization strategy for it that I much prefer